VideoLocker/src/test/java/org/edx/mobile/test/ServerJsonDateAdapterFactoryTest.java

```diff
@@ -0,0 +1,73 @@
+package org.edx.mobile.test;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+
+import org.edx.mobile.core.ServerJsonDateAdapterFactory;
+import org.edx.mobile.util.DateUtil;
+import org.junit.Test;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+public class ServerJsonDateAdapterFactoryTest {
+
+    private Gson newGson() {
+        Gson gson = new GsonBuilder()
+                .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+                .setDateFormat(DateUtil.ISO_8601_DATE_TIME_FORMAT)
+                .registerTypeAdapterFactory(new ServerJsonDateAdapterFactory())
+                .serializeNulls()
+                .create();
+        return gson;
+    }
+
+    @Test
+    public void testParsingNormalDate() {
+        String dateString = "\"2014-11-06T20:16:45Z\"";
+        Gson gson = newGson();
+        Date date = gson.fromJson(dateString, Date.class);
+
+        assertNotNull(date);
+    }
+
+    @Test
+    public void testParsingMicrosecondsDate() {
+        String dateString = "\"2014-11-06T20:16:45.232333Z\"";
+        Gson gson = newGson();
+        Date date = gson.fromJson(dateString, Date.class);
+
+        assertNotNull(date);
+    }
+
+    @Test
+    public void testParsingNonsenseRaises() {
+        String dateString = "foobar";
+        Gson gson = new Gson();
```

Comment:
Should use `newGson()`, otherwise it's not actually going to exercise the `ServerJsonDateAdapterFactory`